libs/template/helpers.go

```diff
@@ -35,6 +35,13 @@ var cachedUser *iam.User
 var cachedIsServicePrincipal *bool
 var cachedCatalog *string
 
+// UUID that is stable for the duration of the template execution. This can be used
+// to populate the `bundle.uuid` field in databricks.yml by template authors.
+//
+// It's automatically logged in our telemetry logs when `databricks bundle init`
+// is run and can be used to attribute DBU revenue to bundle templates.
+var bundleUuid = uuid.New().String()
```

Comment:
To confirm, this line will only run once per bundle initialization? Is that when this class is instantiated?

Reply:
> this line will only run once per bundle initialization?

Yeah. This PR also adds a unit test to confirm this behaviour when materializing the template.

> Is that when this class is instantiated?

There's no class associated with this variable. According to the Go specification the global variables are initialized before the main thread of the go program executes. Specifically it's during the package initialization.

see: https://arc.net/l/quote/vvjgvjzs

Reply:
Can you make this not a global? Looks like it can be scoped to `loadHelpers`.

Reply:
> Can you make this not a global? Looks like it can be scoped to loadHelpers.

We need it to be accessible outside `loadHelpers` helpers as well once we populate it in the telemetry payload.

Also, while `loadHelpers` would work in practice today, if in the future we make multiple loadHelpers function calls to materialize a template we could end up changing the value of the UUID being serialized accidentally

I can switch over to the `cachedBundleUuid *string` pattern like we do for API calls if that's something you'd prefer.